tests/io/test_molfile_reader.py

```diff
@@ -427,3 +476,44 @@ def test_molfile_with_invalid_bond_index_raises_exception(molfile, expected_erro
         match=expected_error_msg,
     ):
         graph_from_molfile_text(molfile)
+
+
+@pytest.mark.parametrize(
+    "molfile, expected_error_msg",
+    [
+        (
+            "\n\n\n  0  0  0     0  0            999 V3000\n"
+            "M  V30 BEGIN CTAB\n"
+            "M  V30 COUNTS 2 1\n"
+            "M  V30 BEGIN ATOM\n"
+            "M  V30 1 * 0 0 0 0\n"
+            "M  V30 2 * 0 0 0 0\n"
+            "M  V30 END ATOM\n"
+            "M  V30 BEGIN BOND\n"
+            "M  V30 1 1 1 2\n"
+            "M  V30 END BOND",
+            re.escape('Two "star" atoms (index 1 and 2) may not be connected'),
+        ),
+        (
+            "\n\n\n  0  0  0     0  0            999 V3000\n"
+            "M  V30 BEGIN CTAB\n"
+            "M  V30 COUNTS 4 1\n"
+            "M  V30 BEGIN ATOM\n"
+            "M  V30 1 * 0 0 0 0\n"
+            "M  V30 2 H 0 0 0 0\n"
+            "M  V30 3 H 0 0 0 0\n"
+            "M  V30 4 H 0 0 0 0\n"
+            "M  V30 END ATOM\n"
+            "M  V30 BEGIN BOND\n"
+            "M  V30 1 1 1 2 ENDPTS=(1 3 4)\n"
```

Comment:
This shouldn't throw an error because it's not missing the closing ")", right?

Reply:
It's raising an exception because it expects ")" after "3". Clearly, the error message is neither very precise nor helpful.

Reply:
This is a deliberate error built in for testing, right? Because the first number in `ENDPTS=(1 3 4)` is strangely not a node but the number of connections ENDPTS is pointing to, so with the one it should be `ENDPTS=(1 3)` and therefore indeed there should be the round bracket after the "3"

Reply:
> This is a deliberate error built in for testing, right?

Yes